Source/Core/VideoBackends/OGL/TextureConverter.cpp

```diff
@@ -242,34 +242,24 @@ static void EncodeToRamUsingShader(GLuint srcTexture, u8* destAddr, u32 dst_line
 
   int dstSize = dst_line_size * dstHeight;
 
-  if ((writeStride != dst_line_size) && (dstHeight > 1))
+  // When the dst_line_size and writeStride are the same, we could use glReadPixels directly to RAM.
+  // But instead we always copy the data via a PBO, because macOS inexplicably prefers this for some
+  // reason.
+  glBindBuffer(GL_PIXEL_PACK_BUFFER, s_PBO);
+  glBufferData(GL_PIXEL_PACK_BUFFER, dstSize, nullptr, GL_STREAM_READ);
+  glReadPixels(0, 0, (GLsizei)(dst_line_size / 4), (GLsizei)dstHeight, GL_BGRA, GL_UNSIGNED_BYTE,
+               nullptr);
+  u8* pbo = (u8*)glMapBufferRange(GL_PIXEL_PACK_BUFFER, 0, dstSize, GL_MAP_READ_BIT);
+
+  for (size_t i = 0; i < dstHeight; ++i)
```

Comment:
Instead of looping here, you could check if dst_line_size == writeStride (which is likely), memcpy() the whole thing in one block, otherwise do it row-by-row.